PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/PluginUtil.java

```
@@ -430,4 +454,136 @@ public static Image getImage(String imagePath) {
+									
+									return;
+								}
+							}
+						}
+					} catch (Exception e) {
+						String errorMsg = "Error installing " + pluginGroupID + "! Please manually install using Eclipse P2 repository from: Help -> Install New Software....";
+						PluginUtil.displayErrorDialogAndLog(getParentShell(), "Fail to install", errorMsg, e);
+					}
+			    }
+			}
+		}
+		
+		String errorMsg = "Error installing " + pluginGroupID + "! In the following installation wizard, please select the right repository and then filter by " + pluginGroupID + "!";
+		PluginUtil.displayErrorDialogAndLog(getParentShell(), "Fail to install", errorMsg, null);
+		provisioningUI.openInstallWizard(null, null, null);
+	}
+	
+	private static URI getEclipseP2Repository() {
+		String repoPrefix = "download.eclipse.org/releases/";
+		
+		ProvisioningUI provisioningUI = ProvisioningUI.getDefaultUI();
+		if (provisioningUI != null) {
+			RepositoryTracker tracker = provisioningUI.getRepositoryTracker();
+			if (tracker != null) {
+				URI[] sites = tracker.getKnownRepositories(provisioningUI.getSession());
+				for (URI site : sites) {
+					if (site.toString().contains(repoPrefix)) {
+						return site;
+					}
+				}
+			}
+		}
+		
+		String start = "Version: ";
+		String end = " (";
+		URI repoSite = null;
+		ProductProperties productProperties = new ProductProperties(Platform.getProduct());
+		if (productProperties != null) {
+			String aboutText = productProperties.getAboutText();
+			int startIndex = aboutText.indexOf(start);
+			int endIndex = aboutText.indexOf(end);
+			String eclipseSimutaneousReleaseVersion = "";
+			if (startIndex >= 0 && endIndex >= 0 && endIndex > startIndex) {
+				eclipseSimutaneousReleaseVersion = aboutText.substring(startIndex + start.length(), endIndex);
+				try {
+					repoSite = new URI("http://" + repoPrefix + eclipseSimutaneousReleaseVersion.toLowerCase());
+				} catch (Exception e) {
+					repoSite = null;
```

The exception can be ignored.